splunk/packages/edit-table/src/EditTable.jsx

```diff
@@ -104,8 +104,23 @@ const EditTable = (props) => {
             }
             return row;
         }
+
+        // when a cell is clicked, the data we get is the label of table header, not the key
```

Comment:
Did you analyze the whole event object to make sure that the key is not hidden somewhere?